### PR TITLE
[WIP]  New Resource: aws_keyspaces_table

### PR DESCRIPTION
**Pull request by saurabh-program**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates  #11221 

Output from acceptance testing:

```
$ make testacc TESTS=TestAccXXX PKG=ec2

...
```

### CODE REVIEW

File: internal/service/keyspaces/find.go [hunk 1]
```diff
@@ -34,3 +34,29 @@ func FindKeyspaceByName(ctx context.Context, conn *keyspaces.Keyspaces, name str
 
 	return output, nil
 }
+
+func FindTableInKeyspaceByName(ctx context.Context, conn *keyspaces.Keyspaces, keyspacceName string, tableName string) (*keyspaces.GetTableOutput, error) {
```
Comment (daryl-d) [inline internal/service/keyspaces/find.go:38]:
```suggestion
func FindTableInKeyspaceByName(ctx context.Context, conn *keyspaces.Keyspaces, keyspaceName string, tableName string) (*keyspaces.GetTableOutput, error) {
```

File: internal/service/keyspaces/find.go [hunk 2]
```diff
@@ -34,3 +34,29 @@ func FindKeyspaceByName(ctx context.Context, conn *keyspaces.Keyspaces, name str
...
+	input := keyspaces.GetTableInput{
+		KeyspaceName: aws.String(keyspacceName),
```
Comment (daryl-d) [inline internal/service/keyspaces/find.go:40]:
```suggestion
		KeyspaceName: aws.String(keyspaceName),
```

File: internal/service/keyspaces/wait.go [hunk 3]
```diff
@@ -60,3 +62,49 @@ func waitKeyspaceDisappears(ctx context.Context, conn *keyspaces.Keyspaces, name
 
 	return err
 }
+
+func waitTableExists(ctx context.Context, conn *keyspaces.Keyspaces, keyspacceName string, tableName string) error {
```
Comment (daryl-d) [inline internal/service/keyspaces/wait.go:66]:
```suggestion
func waitTableExists(ctx context.Context, conn *keyspaces.Keyspaces, keyspaceName string, tableName string) error {
```

File: internal/service/keyspaces/wait.go [hunk 4]
```diff
@@ -60,3 +62,49 @@ func waitKeyspaceDisappears(ctx context.Context, conn *keyspaces.Keyspaces, name
...
+	err := resource.RetryContext(ctx, tableExistsTimeout, func() *resource.RetryError {
+		_, err := FindTableInKeyspaceByName(ctx, conn, keyspacceName, tableName)
```
Comment (daryl-d) [inline internal/service/keyspaces/wait.go:68]:
```suggestion
		_, err := FindTableInKeyspaceByName(ctx, conn, keyspaceName, tableName)
```

File: internal/service/keyspaces/wait.go [hunk 5]
```diff
@@ -60,3 +62,49 @@ func waitKeyspaceDisappears(ctx context.Context, conn *keyspaces.Keyspaces, name
...
+
+		if err != nil {
+			return resource.NonRetryableError(err)
+		}
+
+		return nil
+	})
+
+	if tfresource.TimedOut(err) {
+		_, err = FindTableInKeyspaceByName(ctx, conn, keyspacceName, tableName)
```
Comment (daryl-d) [inline internal/service/keyspaces/wait.go:82]:
```suggestion
		_, err = FindTableInKeyspaceByName(ctx, conn, keyspaceName, tableName)
```

File: internal/service/keyspaces/wait.go [hunk 6]
```diff
@@ -60,3 +62,49 @@ func waitKeyspaceDisappears(ctx context.Context, conn *keyspaces.Keyspaces, name
...
+	}
+
+	return err
+}
+
+func waitTableDisappears(ctx context.Context, conn *keyspaces.Keyspaces, keyspacceName string, tableName string) error {
```
Comment (daryl-d) [inline internal/service/keyspaces/wait.go:88]:
```suggestion
func waitTableDisappears(ctx context.Context, conn *keyspaces.Keyspaces, keyspaceName string, tableName string) error {
```

File: internal/service/keyspaces/wait.go [hunk 7]
```diff
@@ -60,3 +62,49 @@ func waitKeyspaceDisappears(ctx context.Context, conn *keyspaces.Keyspaces, name
...
+	stillExistsErr := fmt.Errorf("keyspace still exists")
+
+	err := resource.RetryContext(ctx, tableDisappearsTimeout, func() *resource.RetryError {
+		_, err := FindTableInKeyspaceByName(ctx, conn, keyspacceName, tableName)
```
Comment (daryl-d) [inline internal/service/keyspaces/wait.go:92]:
```suggestion
		_, err := FindTableInKeyspaceByName(ctx, conn, keyspaceName, tableName)
```

**Comment by ewbankkit** [comment]
Superseded by https://github.com/hashicorp/terraform-provider-aws/pull/24351.

**Pull request closed without merging**